src/pages/single-product/single-product.tsx

```diff
@@ -47,8 +50,18 @@ const SingleProduct = () => {
                 <div className="actions-wrapper">
                   <h1 className="product-title">{product.name}</h1>
                   <div>
-                    <button className="edit-btn btn">Edit</button>
-                    <button className="delete-btn btn" onClick={handleDelete}>
+                    <button
+                      className="edit-btn btn"
+                      onClick={() => navigate("./edit")}
+                      disabled={user?.role !== CustomerRoles.ADMIN}
```

Comment:
It would be better to extract this check in a constant as you use it in more places.

Reply:
Done.